### PR TITLE
build.go: extracts 'fixGoPath()' from huge main()

### PR DESCRIPTION
**Pull request by lkwg82**
### Purpose

Decrease size of main() in build.go

### Testing

go run build.go

### CODE REVIEW

File: build.go [hunk 1]
```diff
@@ -241,6 +231,20 @@ func main() {
 	}
 }
 
+func fixGoPath() {
+	// If GOPATH isn't set, set it correctly with the assumption that we are
+	// in $GOPATH/src/github.com/syncthing/syncthing.
+	if os.Getenv("GOPATH") == "" {
```
Comment (AudriusButkevicius) [inline build.go]: I think we should adhere to the standard Go way of documenting things, which is putting the comment above the function definition starting with `fixGoPath does X Y Z`

**Comment by calmh** [comment]
Yeah... And if we're being pedantic (which we are), it sets the GOPATH it doesn't fix it, and the conditional belongs outside. :)

```
func main() {
    ...
    if os.Getenv("GOPATH") == "" {
        setGoPath()
    }
    ...
}

func setGoPath() {
    // no if here
}
```

File: build.go [hunk 2]
```diff
@@ -241,6 +233,18 @@ func main() {
 	}
 }
 
+// set GOPATH correctly with the assumption that we are
+// in $GOPATH/src/github.com/syncthing/syncthing.
+func setGoPath() {
```
Comment (AudriusButkevicius) [inline build.go]: It needs to start with `setGoPath` as the first word otherwise vet should give a warning.

Comment (lkwg82) [inline build.go]: I dont understand what the issue is. `go vet`seems fine. What do you mean?

Comment (AudriusButkevicius) [inline build.go]: See: https://blog.golang.org/godoc-documenting-go-code

> Notice this comment is a complete sentence that begins with the name of the element it describes.

Comment (lkwg82) [inline build.go]: Maybe we can check this with a tool?

Comment (AudriusButkevicius) [inline build.go]: I think either lint or vet already does that.

**Comment by AudriusButkevicius** [comment]
@st-review merge

build: Extract setGoPath

**Comment by st-review** [comment]
:ok_hand: Merged as 77e47066ed8eabd1aa7b490107af3ff51e935213. Thanks, @lkwg82!

**Pull request closed without merging**